frontend/src/lib/components/Link.tsx

```diff
@@ -16,15 +16,14 @@ export function Link({ to, preventClick = false, tag = 'a', ...props }: LinkProp
 
         if (!props.target) {
             event.preventDefault()
-            event.stopPropagation()
             if (to && to !== '#' && !preventClick) {
                 router.actions.push(to) // router is mounted automatically, so this is safe to call
             }
         }
         props.onClick && props.onClick(event)
     }
 
-    return React.createElement(tag, {
+    return React.createElement(tag as string, {
```

Comment:
This is lying to the compiler, not sure if this works at all :D 

The proper way here seems to be to be use React.cloneElement for non-strings.

Q: Should this be part of this PR?

Reply:
Tagging @yakkomajuri here as he helped with this change. It is definitely unrelated to the PR, but because I made a small change to this file, tests won't pass unless this TS issue is fixed.

Reply:
Yes it is indeed lying to the compiler - But we were passing a type as a `tag` before that wasn't compatible with `createElement` at all. 

Let me take a look at using `cloneElement`

Reply:
`cloneElement` also doesn't accept the types we're using for the button in `LinkButton`: 

`ReactNode` as what we currently have and `CompoundedComponent` the actual type provided by antd

Reply:
Thanks for looking into this @yakkomajuri, how about we keep this workaround for now and add a note to fix this later?

Reply:
If @macobo has time I'd love to learn from him the best approach here. I know you're eager to merge (rightfully so) but I'd like to see if he has a quick fix. Will ping on Slack

Reply:
I think the `correct` way here (if we wanted to support both string and React.Node) is something like the following:

```ts
const props = { ... }
if (typeof tag === 'string') {
  return React.createElement(tag, props)
} else {
  return React.cloneElement(tag, props)
}
```

Not 100% sure this would work though, would need to test it

Reply:
Happy to wait to fix this the right way 😁

Reply:
`cloneElement` doesn't work on the tag being a `CompoundedComponent` or `React.ReactNode`

Reply:
Or alternatively, if tag is a class:

```
return <tag {...props} />
```

Might work

Reply:
Ah so gotta change the callsite

Reply:
let me test

Reply:
Ok so that worked but had to change the button's type to `React.FunctionComponentElement<any>` - any objections @macobo ?